### ISSUE TITLE
Possessive quantifier matches where an ordinary quantifier doesn't

### ISSUE THREAD

**Issue by benrg** [python/cpython#106052]
```python
>>> import re
>>> re.fullmatch('(?:ab?c)*', 'a')
>>> re.fullmatch('(?:ab?c)*+', 'a')
<re.Match object; span=(0, 1), match='a'>
>>>
```

I'm not sure that I completely understand the behavior of the possessive quantifiers, but I think that if `R*` doesn't match a string then `R*+` shouldn't match it, where `R` is any regex.

Versions tested:

    3.12.0b3 (tags/v3.12.0b3:f992a60, Jun 20 2023, 12:25:40) [MSC v.1936 64 bit (AMD64)]
    3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64 bit (AMD64)]
    3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:30:09) [MSC v.1934 32 bit (Intel)]
    3.11.4 (main, Jun 14 2023, 18:33:43) [GCC 10.2.1 20210110]

### Linked PRs
* gh-106515
* gh-107795
* gh-107796

**Comment by ferdnyc** [comment]
If nothing else, the docs claim that

> `x*+`, `x++` and `x?+` are equivalent to `(?>x*)`, `(?>x+)` and `(?>x?)` correspondingly.

And,

```python
>>> import re
>>> re.fullmatch('(?>(?:ab?c)*)', 'a')
>>> 
```

So, that seems a clear discrepancy.

**Comment by arhadthedev** [comment]
@ezio-melotti, @serhiy-storchaka (as re module [experts](https://devguide.python.org/core-developers/experts/#stdlib))

**Comment by serhiy-storchaka** [comment]
It is definitely a bug.

The simplest fix would be implementing possessive quantifiers via atomic groups. It may be less efficient, but gives us time to fix the specialized opcode.

**Comment by serhiy-storchaka** [comment]
#106515 is a last resort. It should be merged if better solution not be found before the next releases.

**Comment by Yhg1s** [comment]
FYI, per comments in #106515, I'm not treating this as a release blocker in 3.12 beta 4.

**Comment by gpshead** [comment]
(per comment on the PR, I don't think this is really a _release blocker_ given the bug shipped in 3.11)

**Comment by serhiy-storchaka** [comment]
It is actually a duplicate of #100061 for which there is a proper fix.
